src/panelSocket.ts

```diff
@@ -102,11 +102,13 @@ export class PanelSocket extends EventEmitter {
         }
     }
 
-    private onClose() {
+    private onClose(message: { wasClean: boolean; code: number; reason: string; target: WebSocket }) {

```

Comment:
You aren't really using this message param (which is actually an event rather than a message).

Let's just remove it and add a real 'closed' event to the PanelSocket that we can emit here.